c/validation/adbc_validation.cc

```diff
@@ -1095,8 +1100,12 @@ void StatementTest::TestSqlIngestBinary() {
       NANOARROW_TYPE_BINARY, {std::nullopt, "", "\x00\x01\x02\x04", "\xFE\xFF"}));
 }
 
+void StatementTest::TestSqlIngestDate32() {
+  ASSERT_NO_FATAL_FAILURE(TestSqlIngestNumericType<int32_t>(NANOARROW_TYPE_DATE32));
```

Comment:
Right now this tests the postgres implementation, but the SQLite implementation doesn't get tested since the roundtrip value is a string.

For the timestamps we have separate test bodies and a function like `ValidateIngestedTimestampData` that the tests use to override what should be returned. We could follow that same pattern here though it is a bit heavy-handed given how this fits into `TestSqlIngestNumericType` pretty easily

The general problem is finding the right pattern to round trip tests for things where the input type doesn't match the output type

Reply:
What about something like this? https://github.com/apache/arrow-adbc/issues/787

What I envision there is a set of directories, one per vendor, where each has a `.sql` file that creates tables and inserts data and a reference `.arrow` file for the expected output.

Reply:
More verbose, but then for drivers where there's overlap (e.g. Postgres in Java, Postgres in C++) we can also use them to assert that the behavior matches. The problem is checking in binaries; I was thinking we could borrow the arrow-testing submodule to do that to avoid cluttering the main repo.